components/heatmapcontrol/heatmapcontrol.js

```diff
@@ -240,3 +316,33 @@ function createOpacities(container, field, changeFunc){
 	container.appendChild(div);
 	return rs;
 }
+//Create HTML Color inputs for given noOfIntervals
+function createIntervalInputs(container, noOfIntervals, changeFunc){
+	//Empty the container
+	while ( container.firstChild ) container.removeChild( container.firstChild );
+	for (let i = 1; i <= noOfIntervals; i++) {	
+	
+		const div = document.createElement('div');
+		div.className = 'color-input-container';
+		const label = document.createElement('label');
+		label.textContent = `Interval ${i} `;
+		label.className = 'color-input'
+		const color = document.createElement('input');
+		color.type = 'color';
+		color.value = defaultColorList[i - 1];
+		color.oninput = changeFunc
+		//Input for color legends.
+		div.appendChild(label);
+		div.appendChild(color);
+	
+		container.appendChild(div);
+	}
+}
+// returns selected colors for intervals
+function getColors(container){
+	const rs = [];
+	$(container).children().each(function (index,colorDiv) {
+		rs.push(colorDiv.querySelector('input').value)
+	});
+	return rs.reverse();
```

Comment:
I'll invite @nanli-emory to comment on this too, but I think it makes more sense to show the lower value heatmap bins first.
```suggestion
	return rs;
```

Reply:
Agreed, i used the reverse order because the code in legend was in that order. I have a feeling that will need to update legend icons code also. I'll check and update accordingly.

Reply:
Also, @birm @nanli-emory what should be maximum limit on number of intervals. As of now i have made min 2 and max 5 intervals.

Reply:
New order
![image](https://user-images.githubusercontent.com/26872044/77674654-ecbe2400-6fb1-11ea-994a-4e3ec55cd16c.png)

Reply:
max 10 intervals

Reply:
The logic for gradient should go from lighter colors for less values to more darker colors for higher values, or from blue to red? 
![image](https://user-images.githubusercontent.com/26872044/77683924-33664b00-6fbf-11ea-8f28-edef0bf6290a.png)

Above screen shot has gradient from white to blue for lower to higher values. Of course user can change color of each interval. If this isn't appropriate please suggest the 10 colors for which gradient should be formed.

Reply:
waiting for your reply @nanli-emory.